### ISSUE TITLE
BUILD: target "grpc++_unsecure" depends on ":grpc" and hence "//external:libssl"

### ISSUE THREAD

**Issue by mrry** [grpc/grpc#7707]
I originally raised this in #6784, and it was fixed by @tberghammer in #6977. However, when #7292 was merged, the spurious dependency was re-added (perhaps accidentally?).

This dependency is preventing us from upgrading TensorFlow to use the latest and greatest version of gRPC, since we only use the unsecure target. What would be the appropriate fix?

**Comment by nicolasnoble** [comment]
Right. This is something we don't have proper testing against, as it requires creating a kind of botched environment that doesn't have any SSL.

#7292 was indeed the wrong solution, as it injects grpc everywhere.

**Comment by nicolasnoble** [comment]
However, #7292 happened on master - the v1.0.x branch should be fine for you to use, and this is where the latest and greatest will happen (soon): #7745.
